### PR TITLE
Ghandi: Fixed Inverse Integer Overflow

### PR DESCRIPTION
**Pull request by HadetTheUndying**
Might need some formatting cleanup. I didn't touch LBA-Slim

Would also like to add for ease of use maybe removing the spaces from the filenames, but that's mostly on my end.

### CODE REVIEW

File: LBA v2.22 Light Deployable.txt [hunk 1]
```diff
@@ -1,339 +1,346 @@
...
+        {
+            key owner=llList2Key(proc,0);
+            string ownern=llList2String(proc,1);
+            key creator=llList2Key(proc,2);
+            string name=llList2Key(proc,3);
+            integer dmg=llList2Integer(proc,4);
+            integer full=dmg;
+            if(atcap>0)
+            {
+                if (atcap) // use atcap to limit repairs
```
Comment (ThunderRahja) [inline LBA v2.22 Light Deployable.txt:229]: Lines 229 to 236: This is not the intended implementation of the fix. These lines should be moved up and inserted after line 183.

File: LBA v2.22 Light Deployable.txt [hunk 2]
```diff
@@ -1,339 +1,346 @@
...
+            {
+                key target=llGetSubString(m,0,35);//Gets the target key from the first 36 (The length of a key) characters in the message
+                if(target==myKey||target==myOwner)//First things first, am I the target?
+                {
+                    key owner=llList2Key(ownerinfo,0);//Gets the owner key from Ownerinfo
+                    key creator=llList2Key(ownerinfo,1);//Gets the creator key from Ownerinfo
+                    integer att=llList2Integer(ownerinfo,2);
+                    string n2=llKey2Name(owner);//Gets the owner name
+                    if(n2=="")n2=owner;//If you can't get the owner name, return the key instead
+                    integer dmg=(integer)llGetSubString(m,37,-1);//This is the damage, fuck you.
```
Comment (ThunderRahja) [inline LBA v2.22 Light Deployable.txt:183]: Here is where the fix should be implemented.

File: LBA v2.22 Light.txt [hunk 3]
```diff
@@ -1,355 +1,362 @@
...
+        {
+            key owner=llList2Key(proc,0);
+            string ownern=llList2String(proc,1);
+            key creator=llList2Key(proc,2);
+            string name=llList2Key(proc,3);
+            integer dmg=llList2Integer(proc,4);
+            integer full=dmg;
+            if(atcap>0)
+            {
+                if (atcap) // use atcap to limit repairs
```
Comment (ThunderRahja) [inline LBA v2.22 Light.txt:229]: Incorrect insertion point.

File: LBA v2.22 Light.txt [hunk 4]
```diff
@@ -1,355 +1,362 @@
...
+                key target=llGetSubString(m,0,35);//Gets the target key from the first 36 (The length of a key) characters in the message
+                if(target==myKey||target==myOwner)//First things first, am I the target?
+                {
+                    key owner=llList2Key(ownerinfo,0);//Gets the owner key from Ownerinfo
+                    key creator=llList2Key(ownerinfo,1);//Gets the creator key from Ownerinfo
+                    integer att=llList2Integer(ownerinfo,2);
+                    string n2=llKey2Name(owner);//Gets the owner name
+                    if(n2=="")n2=owner;//If you can't get the owner name, return the key instead
+                    integer dmg=(integer)llGetSubString(m,37,-1);//This is the damage, fuck you.
+                    if(att)
```
Comment (ThunderRahja) [inline LBA v2.22 Light.txt]: Implement fix here.

File: LBA v2.22 Light.txt [hunk 5]
```diff
@@ -1,355 +1,362 @@
...
+            {
+                key target=llGetSubString(m,0,35);//Gets the target key from the first 36 (The length of a key) characters in the message
+                if(target==myKey||target==myOwner)//First things first, am I the target?
+                {
+                    key owner=llList2Key(ownerinfo,0);//Gets the owner key from Ownerinfo
+                    key creator=llList2Key(ownerinfo,1);//Gets the creator key from Ownerinfo
+                    integer att=llList2Integer(ownerinfo,2);
+                    string n2=llKey2Name(owner);//Gets the owner name
+                    if(n2=="")n2=owner;//If you can't get the owner name, return the key instead
+                    integer dmg=(integer)llGetSubString(m,37,-1);//This is the damage, fuck you.
```
Comment (ThunderRahja) [inline LBA v2.22 Light.txt:183]: Implement fix here.

File: LBA v2.22.txt [hunk 6]
```diff
@@ -1,341 +1,348 @@
...
+        {
+            key owner=llList2Key(proc,0);
+            string ownern=llList2String(proc,1);
+            key creator=llList2Key(proc,2);
+            string name=llList2Key(proc,3);
+            integer dmg=llList2Integer(proc,4);
+            integer full=dmg;
+            if(atcap>0)
+            {
+                if (atcap) // use atcap to limit repairs
```
Comment (ThunderRahja) [inline LBA v2.22.txt:229]: Incorrect.

File: LBA v2.22.txt [hunk 7]
```diff
@@ -1,341 +1,348 @@
...
+            {
+                key target=llGetSubString(m,0,35);//Gets the target key from the first 36 (The length of a key) characters in the message
+                if(target==myKey||target==myOwner)//First things first, am I the target?
+                {
+                    key owner=llList2Key(ownerinfo,0);//Gets the owner key from Ownerinfo
+                    key creator=llList2Key(ownerinfo,1);//Gets the creator key from Ownerinfo
+                    integer att=llList2Integer(ownerinfo,2);
+                    string n2=llKey2Name(owner);//Gets the owner name
+                    if(n2=="")n2=owner;//If you can't get the owner name, return the key instead
+                    integer dmg=(integer)llGetSubString(m,37,-1);//This is the damage, fuck you.
```
Comment (ThunderRahja) [inline LBA v2.22.txt:183]: Insert here.

**Comment by HadetTheUndying** [comment]
Rodger. Thanks for pointing out my dumb

**Pull request closed without merging**